cfme/utils/template/base.py

```diff
@@ -234,19 +241,46 @@ def deploy_template(self):
 
     @log_wrap("download image locally")
     def download_image(self):
+        suffix = re.compile(
+            r'^.*?[.](?P<ext>tar\.gz|tar\.bz2|\w+)$').match(self.image_name).group('ext')
         # Check if file exists already:
         if path.isfile(self.local_file_path):
             logger.info('Local image found, skipping download: %s', self.local_file_path)
-            return True
+            if suffix not in ['zip', 'tar.gz']:
+                return True
+        else:
+            # Download file to cli-tool-client
+            try:
+                request.urlretrieve(self.raw_image_url, self.local_file_path)
+            except URLError:
+                logger.exception('Failed download of image using urllib')
+                return False
 
-        # Download file to cli-tool-client
+        # Unzips image  when suffix is zip or tar.gz and then changes image name to extracted one.
+        # For EC2 and SCVMM images is zip used and for GCE is tar.gz used.
+
+        archive_path = self.image_name
         try:
-            request.urlretrieve(self.raw_image_url, self.local_file_path)
-        except URLError:
-            logger.exception('Failed download of image using urllib')
-            return False
-        else:
+            if suffix == 'zip':
+                archive = ZipFile(archive_path)
+                zipinfo = archive.infolist()
+                self._unzipped_file = zipinfo[0].filename
+            elif suffix == 'tar.gz':
+                archive = tarfile.open(archive_path, "r:gz")
+                self._unzipped_file = archive.firstmember.name
+            else:
+                return True
+            if path.isfile(self.image_name):
+                os.remove(self.image_name)
+            logger.info('Image archived - unzipping as : %s', self._unzipped_file)
+            archive.extractall()
+            archive.close()
+            # remove the archive
+            os.remove(archive_path)
```

Comment:
`archive_path` is equal to `self.image_name`, how is this call different than the one at L274?

Reply:
It isn't equal as image_name returns `return self._unzipped_file or self.raw_image_url.split("/")[-1]` and _unzipped_file is already set. I need to delete original image_name, not image_name after setting _unzipped_file.

Reply:
Also deleting image_name on L274 deletes file that might be already extracted in previous run so it doesn't throw an error that file already exists. so deleting image_name is not equal deleting archive_name.